packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Models/Types/ModelTypeProvider.cs

```diff
@@ -32,7 +37,9 @@ public ModelTypeProvider(InputModelType inputModel, SourceInputModel? sourceInpu
                 DeclarationModifiers |= TypeSignatureModifiers.Abstract;
             }
 
-            _inputModel = inputModel;
+            Usage = inputModel.Usage;
+
```

Comment:
Where we going to remove the serialization ctor from this class and put it in the serialization provider?

Reply:
Yeah I planned to open that as a follow up PR to this one.

Reply:
Here is the draft PR: https://github.com/microsoft/typespec/pull/3410. It is technically ready, but I had started this branch off of this current PR so there is a dependency to this PR.